dolphin/db/sqlalchemy/api.py

```diff
@@ -274,7 +274,15 @@ def _storage_get(context, storage_id, session=None):
 
 
 def _storage_get_query(context, session=None):
-    return model_query(context, models.Storage, session=session)
+    read_deleted = context.read_deleted
+    kwargs = dict()
+
+    if read_deleted in ('no', 'n', False):
```

Comment:
Is there any synchronisation required here? Can there be a race condition

Reply:
It's a read operation, so I think there should be no race condition